dlp/src/main/java/com/example/dlp/RiskAnalysis.java

```diff
@@ -248,83 +336,124 @@ private static void calculateKAnonymity(
               .build();
 
       PrivacyMetric privacyMetric =
-          PrivacyMetric.newBuilder()
-              .setKAnonymityConfig(kanonymityConfig)
-              .build();
+          PrivacyMetric.newBuilder().setKAnonymityConfig(kanonymityConfig).build();
 
-      AnalyzeDataSourceRiskRequest request =
-          AnalyzeDataSourceRiskRequest.newBuilder()
-              .setPrivacyMetric(privacyMetric)
+      String topicName = String.format("projects/%s/topics/%s", projectId, topicId);
+
+      PublishToPubSub publishToPubSub = PublishToPubSub.newBuilder().setTopic(topicName).build();
+
+      // create action to publish job status notifications over Google Cloud Pub/Sub
+      Action action = Action.newBuilder().setPubSub(publishToPubSub).build();
+
+      RiskAnalysisJobConfig riskAnalysisJobConfig =
+          RiskAnalysisJobConfig.newBuilder()
               .setSourceTable(bigQueryTable)
+              .setPrivacyMetric(privacyMetric)
+              .addActions(action)
+              .build();
+
+      CreateDlpJobRequest createDlpJobRequest =
+          CreateDlpJobRequest.newBuilder()
+              .setParent(ProjectName.of(projectId).toString())
+              .setRiskJob(riskAnalysisJobConfig)
+              .build();
+
+      DlpJob dlpJob = dlpServiceClient.createDlpJob(createDlpJobRequest);
+      String dlpJobName = dlpJob.getName();
+
+      final SettableApiFuture<Boolean> done = SettableApiFuture.create();
+
+      // setup a Pub/Sub subscriber to listen on the job completion status
+      Subscriber subscriber =
+          Subscriber.newBuilder(
+              ProjectSubscriptionName.newBuilder()
+                  .setProject(projectId)
+                  .setSubscription(subscriptionId)
+                  .build(),
+              (pubsubMessage, ackReplyConsumer) -> {
+                if (pubsubMessage.getAttributesCount() > 0
+                    && pubsubMessage.getAttributesMap().get("DlpJobName").equals(dlpJobName)) {
+                  // notify job completion
+                  done.set(true);
+                  ackReplyConsumer.ack();
+                }
+              })
               .build();
+      subscriber.startAsync();
+
+      // wait for job completion
+      try{
+        done.get(1, TimeUnit.MINUTES);
+        Thread.sleep(500);
```

Comment:
Why aren't we explaining about alternatives to sleep?

Reply:
Done.